app/models/piece.rb

```diff
@@ -39,6 +39,15 @@ def capturable?(destination_row, destination_column)
     Piece.where(game_id: game.id, row: destination_row, column: destination_column, in_game: true, color: opposite_color).exists?
   end
 
+  def any_move?
+    8.times do |r|
+      8.times do |c|
+        return true if unoccupied_space?(r, c) && valid_move?(r, c)
```

Comment:
Doesn't valid_move? cover the case of unoccupied_space? Seems like you might be able to take unoccupied_space? out of there, though I guess maybe you are doing it that way just to speed things up by not bothering with the other logic if the space is occupied but then again wouldn't unoccupied also return false if there was a capturable piece there? So this could potentially be hiding a valid move.